src/Orleans/Core/GrainAttributes.cs

```diff
@@ -279,12 +279,26 @@ public static FactoryTypes CollectFactoryTypesSpecified<T>()
     [AttributeUsage(AttributeTargets.Class, AllowMultiple=true)]
     public sealed class ImplicitStreamSubscriptionAttribute : Attribute
     {
-        internal string Namespace { get; private set; }
+        public string Namespace { get; private set; }
+
+        public string Provider { get; private set; }
```

Comment:
If stream provider is specified, implicit subscription pub sub system must respect it and only create subscription to this grain for streams from the specified stream provider.  I don't see that change here..

Options:
- Don't allow specification of stream provider name.
- Update Implicit subscription pubsub system to respect specified stream provider name.